src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INCREASE_SIZE = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int index = hash(key);
+        if (table[index] == null) {
+            addNewNode(key, value, index);
+        } else {
+            attachNode(key, value, index);
+        }
+    }
+
+    private int hash(K key) {
```

Comment:
It's not hash, it's bucket index calculated from hashcode
```suggestion
    private int calculateIndexByHashcode(K key) {
```

Reply:
done

Reply:
done